src/web/mjs/connectors/Sukima.mjs

```diff
@@ -0,0 +1,191 @@
+            method: 'POST',
+            body: '{}',
+            headers: {
+                'Content-Type': 'application/json;charset=utf-8',
+            }
+        });
+
+        let chapters = [];
+        let books = await this.fetchJSON(request);
+        for (const book of books.contents) {
+            for (const chapter of book.stories) {
+                chapters.push(
+                    {
+                        id: '/bv/t/' + chapter.title_code + '/v/' + chapter.volume + '/s/' + chapter.story + '/p/0',
+                        title: chapter.volume.toString().padStart(3, '0') + '-' + chapter.story.toString().padStart(3, '0') + ' - ' + chapter.info.text.trim()
+                    }
+                );
+            }
+        }
+        return chapters;
+    }
+
+    async _getPages(manga) {
+        let script = `
+            new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    let shuffle_data = [];
+                    for (page = 0; page < PAGES_INFO.length; page++) {
+                        shuffle_data.push(
+                            {
+                                id: PAGES_INFO[page].page_url,
+                                blocklen: BLOCKLEN,
+                                shuffle_map: JSON.parse(PAGES_INFO[page].shuffle_map)
+                            }
+                        )
+                    }
+                    resolve(shuffle_data);
+                }, 2500);
+            });
+        `;
+        let request = new Request(new URL(manga.id, this.url));
+        let data = await Engine.Request.fetchUI(request, script);
+
+        return data.map(page => {
+            return this.createConnectorURI(
+                {
+                    id: page.id,
+                    blocklen: page.blocklen,
+                    shuffle_map: page.shuffle_map,
+                }
+            );
+        });
+    }
+
+    async _handleConnectorURI(payload) {
+        while (this.lock) {
+            await this.throttle_download( Math.floor(Math.random() * (2000 - 1000)) + 1000);
+        }
+        this.lock = true;
+
+        let request = new Request(new URL(payload.id), this.requestOptions);
+        let response = await fetch(request);
+        let blob = await response.blob();
+        let bitmap = await createImageBitmap(blob);
+
+        let canvas = document.createElement('canvas');
+        canvas.width = bitmap.width;
+        canvas.height = bitmap.height;
+        const ctx = canvas.getContext('2d');
+
+        let xSplitCount = Math.floor(bitmap.width / payload.blocklen);
+        let ySplitCount = Math.floor(bitmap.height / payload.blocklen);
+        let count = 0;
+        ctx.drawImage(bitmap, 0, ySplitCount * payload.blocklen + 1, bitmap.width, bitmap.height - ySplitCount * payload.blocklen, 0, ySplitCount * payload.blocklen, bitmap.width, bitmap.height - ySplitCount * payload.blocklen);
+
+        for (let i = 0; i < xSplitCount; i++) {
```

Comment:
May use more expressive variable name, e.g. `column` instead of `i` and `row` instead of `j`